### DISCUSSION TITLE
No XMLRPC connection to FLRig

### DISCUSSION THREAD

**Discussion by dh0dm** [wavelog/wavelog#1249] [WL Gate]
Hi, 
I ran into a strange problem when I was setting up WLGate. The connection to Wavelog works, I can log WSJTX QSOs via the QSO API. However I was wondering why WLGate does not show up as Radio in Wavelog - so I could not use the Radio API for SSB QSO - no CAT data are shown in Wavelog.
After a while of testing, I used CloudlogBashCat with exact the same settings and API-Key and everything works fine. After that I recognized that WLGate does oviously not connect to my local FlRig (you can see in the FlRig settings if a RPC client is connected).
My settings are default: 127.0.0.1 with port 12345.
Any Ideas?

Thanks in advance, Marko

**Comment by HB9HIL** [comment]
@int2001

**Comment by int2001** [comment]
Hi @dh0dm ,
a few questions, to roundfence the problem, since it isn't reproduceable here:
- Which OS are you using (Since you use CloudlogBashCat, i'd assume it's some linux)?
- if you're using linux - does the launch from the CLI of WLGate shows up connection issues to FLRig?
- depending on your linux-setup: Is there any kind of "Application-Firewall" which possibily prevents the connection from WLGate to FLRig? Sometimes toggling that checkbox may help.
- Did you try to uncheck/check the "FLRig Enabled" checkbox at config within WLGate?
- does WLGate show the latest QRG/Mode, when you turn your VFO?
<img width="205" alt="image" src="https://github.com/user-attachments/assets/98b0070f-8438-4992-9629-32654d814109">

**Reply by dh0dm** [reply]
Hi Joerg,
I am using Arch Linux. On the CLI I see only one message on startup:
`160671:1124/232318.366560:ERROR:sandbox_linux.cc(376)] InitializeSandbox() called with multiple threads in process gpu-process.`

I got a little step further thanks to you comment:
Every time I uncheck and check the FLRig Enabled checkbox, ORG/Mode are updated once on the status pane.
![grafik](https://github.com/user-attachments/assets/4053e955-2c65-4478-b687-7b7b0641839d)
At this time the client connected checkbox ob Flrig turns green for abt a second an is disabled afterwards. So it looks like that WLgate does one single XML-RQ but not subsequent ones. I can verify that on FLRigs Debug Trace for the xml_server code:

```
22:35:46.670 : rig_get_vfo  7160000
22:35:46.672 : mode on  A 
```
There is just that one Request if I (un)check the Checkbox.

73s Marko
